src/components/AuthorizationHOC.jsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { openModalAction } from '../actions';
+
+const AuthorizationHOC = ({ component: Component, token, ...rest }) => (
```

Comment:
Hi @TheDrizzyWay please I'd like to understand how this works

Reply:
The authorization component takes other components as an argument and passes them as props into a route component as long as the user is logged in.